src/main/java/org/javamodularity/moduleplugin/tasks/StartScriptsMutator.java

```diff
@@ -108,6 +109,16 @@ private void updateRelativePath(FileCopyDetails fileCopyDetails) {
         fileCopyDetails.setRelativePath(updatedRelativePath);
     }
 
+    private void updateFinalRunCmnd(CreateStartScripts startScriptsTask) {
+        Path outputDir = startScriptsTask.getOutputDir().toPath();
+        Path bashScript = outputDir.resolve(startScriptsTask.getApplicationName());
+
+        removeClasspathArg(bashScript, "eval set .*", "eval set -- \\$DEFAULT_JVM_OPTS \\$JAVA_OPTS \\$CLI_OPTS \\\"\\$APP_ARGS\\\"");
+
+        Path batFile = outputDir.resolve(startScriptsTask.getApplicationName() + ".bat");
+        removeClasspathArg(batFile, "\"%JAVA_EXE%\" %DEFAULT_JVM_OPTS% %JAVA_OPTS% %CLI_OPTS%.*", "\"%JAVA_EXE%\" %DEFAULT_JVM_OPTS% %JAVA_OPTS% %CLI_OPTS% %CMD_LINE_ARGS%");
```

Comment:
The name of the environment variable for specifying application-specific options (e.g. CLI_OPTS) is not hard-coded.It is instead given by `${optsEnvironmentVar}` for both [unixStartScript.txt](https://github.com/gradle/gradle/blob/a7fb7dd9709d5f3cf1fb8d8750623991018dfb4a/subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt#L181) and [windowsStartScript.txt](https://github.com/gradle/gradle/blob/a7fb7dd9709d5f3cf1fb8d8750623991018dfb4a/subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/windowsStartScript.txt#L86).

By default, the value of `optsEnvironmentVar`  is [derived from the application name](https://github.com/gradle/gradle/blob/702bd8171fda818d771e8fb20768d867aa6d5e81/subprojects/plugins/src/main/java/org/gradle/jvm/application/tasks/CreateStartScripts.java#L124-L134), but it can be also [configured in the Gradle script](https://docs.gradle.org/current/dsl/org.gradle.jvm.application.tasks.CreateStartScripts.html#org.gradle.jvm.application.tasks.CreateStartScripts:optsEnvironmentVar).

Reply:
I'm also a bit concerned that the content of unixStartScript.txt and windowsStartScript.txt may change between different versions of Gradle in a way that breaks our `updateFinalRunCmnd`. But this is not an urgent problem. I found only [one instance of such a breaking change](https://github.com/gradle/gradle/commit/b0fb42928c9d95a389962e2252e8efb0fe0b9fde#diff-c26bc8b96bdceb135484613c52e66d1c) in the past. So, if we decide to address this problem, we can do it in a separate issue. One solution would be to configure our own start script templates instead of using the default ones provided by Gradle.

Reply:
Totally agree with your concern. I really do not like depending on the default scripts. Custom scripts maintained by us is probably the way forward longer term.

Reply:
Nice work, @paulbakker, @siordache! 🙂

PS. If I were to decide, I wouldn't maintain custom scripts in this project (too large cost, too little gain, IMO), and instead would simply count on a bug report should `updateFinalRunCmnd` break in a newer version of Gradle.